### PR TITLE
[Security] [LoginLink] Add technical details

### PR DESCRIPTION
**Pull request by alexandre-daubois**
Fixes https://github.com/symfony/symfony-docs/issues/17139

Although a bit technical, I think it's a good idea to give a bit of details to the reader on this kind of subject. If the reader needs more information and is more curious, the linked classes will be there to guide him through the good parts of the code 👍

### CODE REVIEW

File: security/login_link.rst [hunk 1]
```diff
@@ -428,6 +428,9 @@ The signed URL contains 3 parameters:
     properties. Whenever these change, the hash changes and previous login
     links are invalidated.
 
+For a user that returns ``user@example.com`` on ``$user->getUserIdentifier()``
+call, the generated login link looks like this: ``http://example.com/login_check?user=user@example.com&expires=1675707377&hash=f0Jbda56Y...A5sUCI~TQF701fwJ...7m2n4A~``
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
call, the generated login link looks like this:

.. code-block:: text
    http://example.com/login_check?user=user@example.com&expires=1675707377&hash=f0Jbda56Y...A5sUCI~TQF701fwJ...7m2n4A~
```

?

File: security/login_link.rst [hunk 2]
```diff
@@ -673,6 +676,23 @@ user create this POST request (e.g. by clicking a button)::
         </form>
     {% endblock %}
 
+Hashing strategy
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
Hashing Strategy
```

File: security/login_link.rst [hunk 3]
```diff
@@ -673,6 +676,23 @@ user create this POST request (e.g. by clicking a button)::
...
+Internally, the :class:`Symfony\\Component\\Security\\Http\\LoginLink\\LoginLinkHandler`
+implementation uses the
+:class:`Symfony\\Component\\Security\\Core\\Signature\\SignatureHasher` to create the
+hash contained in the login link.
+
+This hasher creates a first hash with the expiration
+date of the link, the values of the configured signature properties and the
+user identifier. The used hashing algorithm is SHA-256.
+
+Once this first hash is processed and encoded in base 64, a new one is created
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
Once this first hash is processed and encoded in Base64, a new one is created
```

File: security/login_link.rst [hunk 4]
```diff
@@ -673,6 +676,23 @@ user create this POST request (e.g. by clicking a button)::
...
+from the first hash value and the ``kernel.secret`` container parameter. This
+allows Symfony to sign this final hash, which is contained in the login URL.
+This final hash is also a base 64 encoded SHA-256 hash.
```
Comment (OskarStark) [inline security/login_link.rst]:
```suggestion
The final hash is also a Base64 encoded SHA-256 hash.
```

**Comment by alexandre-daubois** [comment]
That's all updated. Thanks for the review and the text code block tip!

File: security/login_link.rst [hunk 5]
```diff
@@ -428,6 +428,12 @@ The signed URL contains 3 parameters:
     properties. Whenever these change, the hash changes and previous login
     links are invalidated.
 
+For a user that returns ``user@example.com`` on ``$user->getUserIdentifier()``
+call, the generated login link looks like this:
+
+.. code-block:: text
```
Comment (OskarStark) [inline security/login_link.rst:434]:
```suggestion
.. code-block:: text

```

**Review by OskarStark — APPROVED** [review]
Its a good first start 👍

**Comment by javiereguiluz** [comment]
I didn't know about these double-hashing details. Thanks Alex for this contribution.

**Pull request merged**